tests/components/test_base_defaults_component.py

```diff
@@ -202,3 +204,62 @@ def test_merge_defaults(self, config: KpopsConfig, handlers: ComponentHandlers):
         )
         assert isinstance(component.nested, Nested)
         assert component.nested == Nested(**{"foo": "foo", "bar": False})
+
+    @pytest.mark.parametrize(
+        ("pipeline_path", "environment", "expected_default_paths"),
+        [
+            (
+                RESOURCES_PATH
+                / "pipelines/test-distributed-defaults/pipeline-deep/pipeline.yaml",
+                "development",
+                [
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/pipeline-deep/defaults.yaml"
+                    ),
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/defaults_development.yaml"
+                    ),
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/defaults.yaml"
+                    ),
+                    Path(f"{RESOURCES_PATH}/defaults_development.yaml"),
+                    Path(f"{RESOURCES_PATH}/defaults.yaml"),
+                ],
+            ),
+            (
+                RESOURCES_PATH
+                / "pipelines/test-distributed-defaults/pipeline-deep/pipeline.yaml",
+                "production",
+                [
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/pipeline-deep/defaults_production.yaml"
+                    ),
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/pipeline-deep/defaults.yaml"
+                    ),
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/defaults_production.yaml"
+                    ),
+                    Path(
+                        f"{RESOURCES_PATH}/pipelines/test-distributed-defaults/defaults.yaml"
+                    ),
+                    Path(f"{RESOURCES_PATH}/defaults.yaml"),
+                ],
+            ),
+        ],
+    )
+    def test_get_defaults_file_paths(
+        self,
+        pipeline_path: Path,
+        environment: str,
```

Comment:
```suggestion
        environment: str | None,
```
please add a test for the `None` case